### ISSUE TITLE
Lightweight install option?

### ISSUE THREAD

**Issue by KyleKotowick** [optuna/optuna#3035]
I'd like to use Optuna for a project, but the installed package with dependencies comes to a whopping 174MB. This is an issue for my application since I'd like to use it in AWS Lambda, which has a hard cap of 250MB for all code in a lambda, and some other packages I'm using consume 90MB (bringing me to 260MB).

Is there a lightweight install option for Optuna, a way to strip out any optional functionality? Or any other suggestions on how to get the size down?

Thanks!

**Comment by himkt** [comment]
Hello @KyleKotowick, thank you for opening the issue. And sorry for the late response.

```sh

> docker run --rm --platform linux/amd64 -it python:3.9.7 bash
root:/usr/local/lib/python3.9/site-packages# pip install optuna
...
root:/usr/local/lib/python3.9/site-packages# du -h -d 1| sort -h
...
400K	./stevedore
472K	./tqdm
492K	./wcwidth
596K	./mako
616K	./greenlet
636K	./cliff
736K	./pyparsing
804K	./pbr
940K	./pkg_resources
1.1M	./cmd2
1.9M	./alembic
2.5M	./yaml
2.8M	./optuna
3.0M	./setuptools
13M	./sqlalchemy
14M	./pip
33M	./numpy
34M	./scipy.libs
35M	./numpy.libs
110M	./scipy
255M	.
```

Unfortunately, it is difficult to reduce the size of Optuna for now.
We depend on NumPy and SciPy as requirements, which consumes about 200MB on the storage.
It could be possible to make SciPy optional but we don't have currently any plan to do so.

**Comment by not522** [comment]
This issue has been resolved by #4105.

```
root@a0039ce61c4c:/usr/local/lib/python3.9/site-packages# du -h -d 1| sort -h
16K	./_distutils_hack
16K	./_yaml
28K	./MarkupSafe-2.1.1.dist-info
28K	./PyYAML-6.0.dist-info
32K	./wheel-0.37.0.dist-info
36K	./Mako-1.2.4.dist-info
36K	./colorlog-6.7.0.dist-info
40K	./packaging-23.0.dist-info
44K	./alembic-1.9.1.dist-info
44K	./cmaes-0.9.1.dist-info
44K	./greenlet-2.0.1.dist-info
48K	./colorlog
56K	./setuptools-57.5.0.dist-info
68K	./SQLAlchemy-1.4.46.dist-info
72K	./optuna-3.1.0.dev0.dist-info
88K	./tqdm-4.64.1.dist-info
96K	./markupsafe
96K	./pip-21.2.4.dist-info
120K	./cmaes
160K	./wheel
192K	./numpy-1.24.1.dist-info
276K	./packaging
488K	./tqdm
516K	./pkg_resources
648K	./mako
1.9M	./alembic
2.1M	./greenlet
2.3M	./setuptools
2.5M	./yaml
3.3M	./optuna
9.7M	./pip
13M	./sqlalchemy
35M	./numpy
35M	./numpy.libs
106M	.
```
